cloudsupport/cloudproviderconfiguration_test.go

```
@@ -0,0 +1,90 @@
+// unsetEnvWithCleanup unsets an env var for the duration of the test and
+// restores its original value (or absence) via t.Cleanup, so concurrent or
+// subsequent tests are not affected.
+func unsetEnvWithCleanup(t *testing.T, key string) {
+	t.Helper()
+	prev, had := os.LookupEnv(key)
+	if err := os.Unsetenv(key); err != nil {
+		t.Fatalf("unset %s: %v", key, err)
+	}
+	t.Cleanup(func() {
+		if had {
+			os.Setenv(key, prev)
```

golangci-lint (2.11.4)

[error] Error return value of `os.Setenv` is not checked

(errcheck)